rcsb/utils/dictionary/DictMethodEntryHelper.py

```diff
@@ -621,18 +623,22 @@ def consolidateAccessionDetails(self, dataContainer, catName, **kwargs):
         try:
             logger.debug("Starting with  %r %r %r", dataContainer.getName(), catName, kwargs)
             #
-            # Add missing pdbx_database_status for MA models
+            # Add missing pdbx_database_status for MA or AF models (if absent in mmCIF file)
             cName = "pdbx_database_status"
-            if dataContainer.exists("entry"):
-                dObj = dataContainer.getObj("entry")
-                entryId = dObj.getValue("id", 0)
-                if entryId.startswith("ma"):
-                    if not dataContainer.exists(cName):
-                        dataContainer.append(DataCategory(cName, attributeNameList=self.__dApi.getAttributeNameList(cName)))
-                        eObj = dataContainer.getObj(cName)
-                        eObj.setValue(entryId, "entry_id", 0)
-                        eObj.setValue("REL", "status_code", 0)
-                        eObj.setValue("?", "recvd_initial_deposition_date", 0)
+            if not dataContainer.exists(cName):
+                if dataContainer.exists("entry") and dataContainer.exists("ma_data"):
+                    dObj = dataContainer.getObj("entry")
+                    entryId = dObj.getValue("id", 0)
+                    compModelSourceId = None
+                    if entryId.upper().startswith("MA_") or entryId.upper().startswith("MA-") or entryId.upper().startswith("AF_"):
+                        compModelSourceId = self.__mcP.getCompModelData(entryId)["sourceId"]
+                        logger.debug("compModelSourceId %r for entryId %s", compModelSourceId, entryId)
+                        if compModelSourceId:
+                            dataContainer.append(DataCategory(cName, attributeNameList=self.__dApi.getAttributeNameList(cName)))
+                            eObj = dataContainer.getObj(cName)
+                            eObj.setValue(compModelSourceId, "entry_id", 0)
+                            eObj.setValue("REL", "status_code", 0)
+                            eObj.setValue("?", "recvd_initial_deposition_date", 0)
             # if there is incomplete accessioninformation then exit
             if not (dataContainer.exists("pdbx_database_status") and dataContainer.exists("pdbx_audit_revision_history")):
```

Comment:
Is line 643 (637 in the previous version) needs to be fixed. `rcsb_accession_info` is not getting populated for MA models because `pdbx_audit_revision_history` is missing for these models